### PR TITLE
Add config and registry to processed source if the methods exist

### LINKED ISSUE #76: Pronto raises cop errors since rubocop v1.38.0
When running `pronto run`, I get the following error messages:
````
An error occurred while Bundler/OrderedGems cop was inspecting /<project-path>/Gemfile.
To see the complete backtrace run rubocop -d.
An error occurred while Lint/MissingCopEnableDirective cop was inspecting /<project-path>/Gemfile.
To see the complete backtrace run rubocop -d.
An error occurred while Lint/RedundantCopEnableDirective cop was inspecting /<project-path>/Work/aep-w2/Gemfile.
To see the complete backtrace run rubocop -d.
````

However, when running `rubocop -d Gemfile` I get no such errors. 
This happens with rubocop `v1.38.0`. When I downgrade to an earlier version, these error messages disappear.

### PR DESCRIPTION
**Pull request by renatolond**
Fixes issue with rubocop > 1.38.x

It seems that https://github.com/rubocop/rubocop/pull/10987 introduced registry and config to new versions of rubocop which were not there before.
Without those, rubocop still works, but it raises a lot of errors for some files.

I added the registry and source like in the rubocop code if the ProcessedSource responds to it, so I'm hoping it won't break compatibility with older rubocop versions.

This fixes #76

### CODE REVIEW

**Review by ashkulz — CHANGES_REQUESTED** [review]
@renatolond can you specifically add `1.38.0` to [this line](https://github.com/prontolabs/pronto-rubocop/blob/v0.11.2/.github/workflows/checks.yml#L18) so that we check this? You'll probably need to add to the excludes as well for Ruby 2.3 and 2.4.

I'm also wondering why I don't get an option to trigger the CI run on this PR :thinking:

**Comment by renatolond** [comment]
> @renatolond can you specifically add `1.38.0` to [this line](https://github.com/prontolabs/pronto-rubocop/blob/v0.11.2/.github/workflows/checks.yml#L18) so that we check this? You'll probably need to add to the excludes as well for Ruby 2.3 and 2.4.
> 
> I'm also wondering why I don't get an option to trigger the CI run on this PR thinking

Sure, done!

It's weird about the CI, I already saw the option of "allow run" for new contributors on some of my projects, but not showing at all is odd :thinking:

**Review by ashkulz — APPROVED** [review]

**Comment by ashkulz** [comment]
@renatolond can you do a rebase or something? I triggered a [run manually](https://github.com/prontolabs/pronto-rubocop/actions/runs/3419876230), apparently [scheduled actions got suspended](https://stackoverflow.com/q/67184368) on this repo, which were supposed to inform me about regressions -- another fun thing to research :see_no_evil:

**Comment by renatolond** [comment]
@ashkulz Sure, done! :)

File: lib/pronto/rubocop/patch_cop.rb [hunk 1]
```diff
@@ -22,10 +22,15 @@ def messages
...
-          path,
-          rubocop_config.target_ruby_version
-        )
+        @processed_source ||= begin
+          processed_source = ::RuboCop::ProcessedSource.from_file(
+            path,
+            rubocop_config.target_ruby_version
+          )
+          processed_source.registry = registry if processed_source.responds_to(:registry=)
+          processed_source.config = rubocop_config if processed_source.responds_to(:config=)
```
Comment (ashkulz) [inline lib/pronto/rubocop/patch_cop.rb:31]:
```suggestion
          processed_source.registry = registry if processed_source.respond_to?(:registry=)
          processed_source.config = rubocop_config if processed_source.respond_to?(:config=)
```

**Review by ashkulz — CHANGES_REQUESTED** [review]
Looks like there's a typo, @renatolond :slightly_smiling_face:

File: lib/pronto/rubocop/patch_cop.rb (hunk 1 above)
Comment (renatolond) [reply lib/pronto/rubocop/patch_cop.rb]: Fixed!

Comment (ashkulz) [reply lib/pronto/rubocop/patch_cop.rb]: I think you made another typo, it's [`respond_to?`](https://ruby-doc.org/core-2.7.6/Object.html#method-i-respond_to-3F) :disappointed:

Comment (renatolond) [reply lib/pronto/rubocop/patch_cop.rb]: sorry, let me run the tests locally before doing another push :S

**Review by ashkulz — APPROVED** [review]

**Comment by ashkulz** [comment]
Thanks for the contribution, @renatolond! I'll push up a release by EOD.

**Comment by renatolond** [comment]
> Thanks for the contribution, @renatolond! I'll push up a release by EOD.

Thank you, @ashkulz :tada:

**Comment by ashkulz** [comment]
Released to RubyGems: https://rubygems.org/gems/pronto-rubocop/versions/0.11.3

**Pull request merged**